File: CS/MasterReport.cs
using DevExpress.XtraReports.UI;
// ...

namespace SideBySideReport {
    public partial class MasterReport : DevExpress.XtraReports.UI.XtraReport {
        public MasterReport() {
            InitializeComponent();
        }

        private void xrSubreport1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            ((XRSubreport)sender).ReportSource.FilterString = "[EmployeeID] = 1";
        }

        private void xrSubreport2_BeforePrint_1(object sender, System.Drawing.Printing.PrintEventArgs e) {
            ((XRSubreport)sender).ReportSource.FilterString = "[EmployeeID] = 2";
        }
    }
}

File: VB/MasterReport.vb
Imports DevExpress.XtraReports.UI
' ...

Public Class MasterReport

    Private Sub xrSubreport1_BeforePrint(ByVal sender As Object, ByVal e _
    As System.Drawing.Printing.PrintEventArgs) Handles XrSubreport1.BeforePrint
        CType(sender, XRSubreport).ReportSource.FilterString = "[EmployeeID] = 1"
    End Sub

    Private Sub xrSubreport2_BeforePrint(ByVal sender As Object, ByVal e _
    As System.Drawing.Printing.PrintEventArgs) Handles XrSubreport2.BeforePrint
        CType(sender, XRSubreport).ReportSource.FilterString = "[EmployeeID] = 2"
    End Sub
End Class